ui/pages/confirm-transaction/confirm-transaction.component.js

```diff
@@ -54,20 +54,36 @@ const ConfirmTransaction = () => {
   const [isMounted, setIsMounted] = useState(false);
   const [pollingToken, setPollingToken] = useState();
 
+  const [transactionId, setTransactionId] = useState();
+  const [isValidERC20TokenMethod, setIsValidERC20TokenMethod] = useState(false);
+  const [isValidTransactionId, setIsValidTransactionId] = useState(false);
+
   const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
   const sendTo = useSelector(getSendTo);
   const unapprovedTxs = useSelector(getUnapprovedTransactions);
   const unconfirmedTxs = useSelector(unconfirmedTransactionsListSelector);
   const unconfirmedMessages = useSelector(unconfirmedTransactionsHashSelector);
 
-  const totalUnapproved = unconfirmedTxs.length || 0;
+  const totalUnapproved = useMemo(() => {
+    return unconfirmedTxs.length || 0;
+  }, [unconfirmedTxs.length]);
 
   const transaction = useMemo(() => {
-    return totalUnapproved
+    const tx = totalUnapproved
       ? unapprovedTxs[paramsTransactionId] ||
-          unconfirmedMessages[paramsTransactionId] ||
-          unconfirmedTxs[0]
+        unconfirmedMessages[paramsTransactionId] ||
+        unconfirmedTxs[0]
       : {};
+
+    const txId = tx.id && String(tx.id);
+
+    setTransactionId(txId);
+    setIsValidERC20TokenMethod(isTokenMethodAction(tx.type));
```

Comment:
This can be `setIsValidTokenMethod` as it check for all token types.

Reply:
I was confused about this because the description of `isTokenMethodAction` says

>  * Given an transaction category, returns a boolean which indicates whether the transaction is calling an erc20 token method

I can change it back